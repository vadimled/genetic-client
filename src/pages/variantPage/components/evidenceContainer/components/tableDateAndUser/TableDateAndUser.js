import React from "react";
import PropTypes from "prop-types";
import style from "./TableDateAndUser.module.scss";

function TableDateAndUser({ date, user }) {
  const
    options = { day: "2-digit", month: "short", year: "numeric" },
    fDate = new Date(date),
    formattedDate = fDate
      .toLocaleDateString("en-GB", options)
      .split(" ")
      .join("/");

  return (
    <div className={style["table-date-and-user-wrapper"]}>
      <div className="table-date-and-user">
        <div className="date">{formattedDate}</div>
        <div className="user">{user.name}</div>
      </div>
    </div>
  );
}

TableDateAndUser.propTypes = {
  date: PropTypes.string,
  user: PropTypes.object
};

export default TableDateAndUser;
