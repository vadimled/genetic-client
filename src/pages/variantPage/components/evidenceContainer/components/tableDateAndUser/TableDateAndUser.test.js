import React from "react";
import { renderWithRedux } from "Utils/test_helpers";
import "jest-dom/extend-expect";
import TableDateAndUser from "./TableDateAndUser";

describe("<TableDateAndUser />", () => {
  it("snapshot", () => {
    const { asFragment } = renderWithRedux(
      <TableDateAndUser
        date={"2019-08-12T05:54:18.074Z"}
        user={{ user: "Hetro " }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
