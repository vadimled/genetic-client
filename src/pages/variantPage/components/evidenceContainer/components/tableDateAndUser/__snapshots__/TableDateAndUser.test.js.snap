// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableDateAndUser /> snapshot 1`] = `
<DocumentFragment>
  <div
    class="table-date-and-user-wrapper"
  >
    <div
      class="table-date-and-user"
    >
      <div
        class="date"
      >
        date
      </div>
      <div
        class="user"
      >
        Name
      </div>
    </div>
  </div>
</DocumentFragment>
`;
